tests/data/collapse-dtrace/results/stack-ustack.txt

```diff
@@ -0,0 +1,4 @@
+libc.so.1`_lwp_start;libc.so.1`_thrp_setup;cru-ds-fdsync-v1`std::sys::unix::thread::Thread::new::thread_start;cru-ds-fdsync-v1`core::ops::function::FnOnce::call_once{{vtable.shim}};cru-ds-fdsync-v1`std::sys_common::backtrace::__rust_begin_short_backtrace;cru-ds-fdsync-v1`rayon_core::registry::ThreadBuilder::run;cru-ds-fdsync-v1`rayon_core::registry::WorkerThread::wait_until_cold;cru-ds-fdsync-v1`<rayon_core::job::HeapJob<BODY> as rayon_core::job::Job>::execute;cru-ds-fdsync-v1`std::panicking::try;cru-ds-fdsync-v1`crucible_downstairs::extent::Extent::flush;cru-ds-fdsync-v1`<crucible_downstairs::extent_inner_raw::RawInner as crucible_downstairs::extent::ExtentInner>::flush;cru-ds-fdsync-v1`std::fs::File::sync_all;libc.so.1`__fdsync;unix`sys_syscall;genunix`fdsync;genunix`fop_fsync;zfs`zfs_fsync;zfs`zil_commit;zfs`zil_commit_impl;zfs`zil_commit_itx_assign;zfs`zil_itx_assign;zfs`dmu_objset_ds 1
```

Comment:
I wonder if we should do something to annotate that these come from calls in the kernel so that they can be given a different color in the flamegraph output given we have that information from DTrace? Feels like a shame to throw it away. What do you think?

Reply:
There's certainly a bunch of data in here that we're not using, but I'd worry about overspecializing for a particular system.

My ideal system would be something like a config file that specifies regex → palette, or even a small DSL for generating a color.

```json
[
  {
    "regex": "zfs`",
    "palette": "warm"
  }, 
  {
    "regex": "libc`",
    "palette": "cool"
  },
  {
    "regex": ".*::.*",
    "palette": "green"
  }
  ...etc
]
```

The language-specific color schemes could be represented with this format, but users could also bring their own config file for flexibility (without having to edit `inferno` itself)

Anyways, this is probably orthogonal to the PR 😄